src/videotile/DefaultVideoTile.ts

```diff
@@ -78,10 +78,15 @@ export default class DefaultVideoTile implements DevicePixelRatioObserver, Video
         mediaStream.removeTrack(track);
       }
 
-      // Need to yield the message loop before clearing `srcObject` to
-      // prevent Safari from crashing.
-      new AsyncScheduler().start(() => {
-        videoElement.srcObject = null;
+      const clearSrcCallBack = ((originalSource: MediaStream | MediaSource | Blob | null) => {
+        return () => {
+          if (videoElement.srcObject === originalSource) {
+            videoElement.srcObject = null;
+          }
+        };
+      })(videoElement.srcObject);
+      requestAnimationFrame(() => {
```

Comment:
Have you tested what happens when the browser tab is backgrounded? For power savings reasons, `requestAnimationFrame` does not tick when the page is hidden (like when you switch tabs in the browser).”